scripts/firedrake-install

```diff
@@ -502,6 +502,9 @@ else:
 
 if options.get("mpiexec") is not None:
     petsc_options.add("--with-mpiexec={}".format(options["mpiexec"]))
+    petsc_options.add("--with-cc={}".format(options["mpicc"]))
+    petsc_options.add("--with-cxx={}".format(options["mpicxx"]))
+    petsc_options.add("--with-fc={}".format(options["mpif90"]))
```

Comment:
All (or none) of `args.mpicc and args.mpicxx and args.mpif90 and args.mpiexec` must be specified when invoking `firedrake-install`, is there any harm in passing them all to PETSc configure?

This ensures the correct compilers are used for all the built packages.